beacon-chain/core/helpers/rewards_penalties.go

```diff
@@ -83,3 +83,67 @@ func InactivityPenalty(
 	validatorBalance := EffectiveBalance(state, validatorIndex)
 	return baseReward + validatorBalance*epochsSinceFinality/params.BeaconConfig().InactivityPenaltyQuotient/2
 }
+
+// GetBalance returns the Gwei balance of a validator by its index
+//
+// Spec pseudocode definition:
+//   def get_balance(state: BeaconState, index: ValidatorIndex) -> Gwei:
+//     """
+//     Return the balance for a validator with the given ``index``.
+//     """
+//     return state.balances[index]
+func GetBalance(state *pb.BeaconState, validatorIndex uint64) uint64 {
+	return state.ValidatorBalances[validatorIndex]
+}
+
+// SetBalance sets the Gwei balance of a validator by its index
+//
+// Spec pseudocode definition:
+// def set_balance(state: BeaconState, index: ValidatorIndex, balance: Gwei) -> None:
+//     """
+//     Set the balance for a validator with the given ``index`` in both ``BeaconState``
+//     and validator's rounded balance ``high_balance``.
+//     """
+//     validator = state.validator_registry[index]
+//     HALF_INCREMENT = HIGH_BALANCE_INCREMENT // 2
+//     if validator.high_balance > balance or validator.high_balance + 3 * HALF_INCREMENT < balance:
+//         validator.high_balance = balance - balance % HIGH_BALANCE_INCREMENT
+//     state.balances[index] = balance
+func SetBalance(state *pb.BeaconState, validatorIndex uint64, balance uint64) {
```

Comment:
Same, I think we'd prefer `state.ValidatorRegistry[idx]` over `state.ValidatorRegistry[validatorIndex]`

```suggestion
func SetBalance(state *pb.BeaconState, idx uint64, balance uint64) {
```

Reply:
thanks, done